Frameworks/Foundation/NSFileManager.mm

```diff
@@ -771,11 +889,86 @@ - (id)attributesOfFileSystemForPath:(id)pathAddr error:(NSError**)error {
 }
 
 /**
- @Status Stub
-*/
-- (BOOL)setAttributes:(id)attribs ofItemAtPath:(id)pathAddr error:(NSError**)err {
-    UNIMPLEMENTED();
-    return YES;
+ @Status Caveat
+ @Notes Only NSFileType, NSFileImmutable are supported
+        NSFileSize, NSFileModificationDate, and NSFileCreationDate cannot be set using WRL APIs
+*/
+- (BOOL)setAttributes:(NSDictionary<NSFileAttributeKey, id>*)attribs ofItemAtPath:(NSString*)path error:(NSError**)error {
+    // TODO:: GH____ convert to C++ winrt when SavePropertiesAsync is fixed
+    ComPtr<ABI::Windows::Storage::IStorageFileStatics> storageFileStatics;
+    RETURN_FALSE_IF_FAILED(Windows::Foundation::GetActivationFactory(
+        Microsoft::WRL::Wrappers::HStringReference(RuntimeClass_Windows_Storage_StorageFile).Get(), &storageFileStatics));
+
+    NSString* filePath = [path stringByReplacingOccurrencesOfString:@"/" withString:@"\\"];
+    ComPtr<ABI::Windows::Foundation::IAsyncOperation<ABI::Windows::Storage::StorageFile*>> fileOperation;
+    RETURN_FALSE_IF_FAILED(storageFileStatics->GetFileFromPathAsync(Strings::NarrowToWide<HSTRING>(filePath).Get(), &fileOperation));
+
+    ComPtr<ABI::Windows::Storage::IStorageFile> file;
+    RETURN_FALSE_IF_FAILED(WRLHelpers::AwaitOperationResult(fileOperation.Get(), &file));
+
+    ComPtr<ABI::Windows::Storage::IStorageItem> fileAsItem;
+    RETURN_FALSE_IF_FAILED(file.As(&fileAsItem));
+
+    ABI::Windows::Storage::FileAttributes fileAttributes;
+    RETURN_FALSE_IF_FAILED(fileAsItem->get_Attributes(&fileAttributes));
```

Comment:
I'm moving the file getting part to a helper, but don't want to for the attributes bit since I need the file too